Cover summary builders with unit tests

**Description**

We need to cover  `SimpleCommentBuilder.kt`, `SimpleNameBuilder.kt`, and `SimpleClusterBuilder.kt` with unit tests to capture the current behavior and make it easier to catch issues in the future.

I will try to do it using UTBot and on my own as well.

UTBot also failed to generate tests for `SimpleNameBuilder.kt`
Error report:
~~~
    ///region Errors report for conditionStep

    public void testConditionStep_errors() {
        // Couldn't generate some tests. List of errors:
        // 
        // 6 occurrences of:
        // Default concrete execution failed

        // 4 occurrences of:
        /* Error in the child process |> org.mockito.exceptions.base.MockitoException: 	Mockito cannot mock
        this class: class org.utbot.framework.plugin.api.Step.	Can not mock final classes with the
        following settings :	 - explicit serialization (e.g. withSettings().serializable())	 - extra
        interfaces (e.g. withSettings().extraInterfaces(...))		You are seeing this disclaimer because Mockito is
        configured to create inlined mocks.	You can learn about inline mocks
        and their limitations under item #39 of the Mockito class
        javadoc.		Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [class
        org.utbot.framework.plugin.api.Step]
        		at org.utbot.framework.concrete.MockValueConstructor.generateMockitoMock(MockValueConstructor.kt:220)
        		at org.utbot.framework.concrete.MockValueConstructor.constructObject(MockValueConstructor.kt:166)
        		at org.utbot.framework.concrete.MockValueConstructor.construct(MockValueConstructor.kt:127)
        		at org.utbot.framework.concrete.MockValueConstructor.constructMethodParameters(MockValueConstructor.kt:104)
        		at org.utbot.framework.concrete.UtExecutionInstrumentation.invoke(UtExecutionInstrumentation.kt:144)
        		at org.utbot.framework.concrete.UtExecutionInstrumentation.invoke(UtExecutionInstrumentation.kt:107)
        		at org.utbot.instrumentation.process.ChildProcessKt.loop(ChildProcess.kt:133)
        		at org.utbot.instrumentation.process.ChildProcessKt.main(ChildProcess.kt:77)
        		at org.utbot.instrumentation.process.ChildProcessKt.main(ChildProcess.kt)
        	Caused by: org.mockito.exceptions.base.MockitoException: Could not modify all classes [class org.utbot.framework.plugin.api.Step]
        		at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:158)
        		at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:372)
        		at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:180)
        		at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:383)
        		... 9 more
        	Caused by: java.lang.IllegalStateException: 	Byte Buddy could not instrument all classes
        within the mock's type hierarchy		This problem should never occur for
        javac-compiled classes. This problem has been observed for classes that
        are:	 - Compiled by older versions of scalac	 - Classes
        that are part of the Android distribution
        		at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:280)
        		at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.mockClass(InlineBytecodeGenerator.java:213)
        		at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.lambda$mockClass$0(TypeCachingBytecodeGenerator.java:47)
        		at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:158)
        		at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:372)
        		at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:180)
        		at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:383)
        		at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:40)
        		at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createMockType(InlineDelegateByteBuddyMockMaker.java:389)
        		at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.doCreateMock(InlineDelegateByteBuddyMockMaker.java:349)
        		at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createMock(InlineDelegateByteBuddyMockMaker.java:328)
        		at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.createMock(InlineByteBuddyMockMaker.java:56)
        		at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:53)
        		at org.mockito.internal.MockitoCore.mock(MockitoCore.java:96)
        		at org.mockito.Mockito.mock(Mockito.java:1964)
        		at org.mockito.Mockito.mock(Mockito.java:1937)
        		... 9 more
        	Caused by: java.lang.NoClassDefFoundError: Lsoot/jimple/Stmt;
        		at java.lang.Class.getDeclaredFields0(Native Method)
        		at java.lang.Class.privateGetDeclaredFields(Class.java:2611)
        		at java.lang.Class.getDeclaredFields(Class.java:1944)
        		at net.bytebuddy.description.type.TypeDescription$ForLoadedType.getDeclaredFields(TypeDescription.java:8797)
        		at net.bytebuddy.dynamic.scaffold.InstrumentedType$Factory$Default$1.represent(InstrumentedType.java:437)
        		at net.bytebuddy.ByteBuddy.redefine(ByteBuddy.java:867)
        		at net.bytebuddy.ByteBuddy.redefine(ByteBuddy.java:842)
        		at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.transform(InlineBytecodeGenerator.java:382)
        		at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
        		at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
        		at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
        		at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
        		at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:276)
        		... 24 more
        	Caused by: java.lang.ClassNotFoundException: soot.jimple.Stmt
        		at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
        		at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        		at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        		... 37 more
        	 */

    }
~~~

--- a/utbot-summary/build.gradle
+++ b/utbot-summary/build.gradle
@@ -9,5 +9,7 @@ dependencies {
 
     implementation group: 'com.github.javaparser', name: 'javaparser-core', version: '3.22.1'
 
+    testImplementation 'org.mockito:mockito-core:4.7.0'
+
     testImplementation("org.junit.jupiter:junit-jupiter:$junit5_version")
 }

--- a/utbot-summary/build.gradle
+++ b/utbot-summary/build.gradle
@@ -9,7 +9,7 @@ dependencies {
 
     implementation group: 'com.github.javaparser', name: 'javaparser-core', version: '3.22.1'
 
-    testImplementation 'org.mockito:mockito-core:4.7.0'
+    testImplementation group: 'org.mockito', name: 'mockito-core', version: mockito_version
 
     testImplementation("org.junit.jupiter:junit-jupiter:$junit5_version")
 }

--- a/utbot-summary/src/main/kotlin/org/utbot/summary/comment/SimpleCommentBuilder.kt
+++ b/utbot-summary/src/main/kotlin/org/utbot/summary/comment/SimpleCommentBuilder.kt
@@ -349,7 +349,7 @@ open class SimpleCommentBuilder(
      * In case when an enclosing class in nested, we need to replace '$' with '.'
      * to render the reference.
      */
-    protected fun invokeDescription(className: String, methodName: String, methodParameterTypes: List<Type>): String {
+    fun invokeDescription(className: String, methodName: String, methodParameterTypes: List<Type>): String {
         val prettyClassName: String = className.replace("$", ".")
 
         return if (methodParameterTypes.isEmpty()) {

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt
@@ -1,6 +1,6 @@
 package org.utbot.summary.comment
 
-import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
@@ -45,13 +45,18 @@ class SimpleClusterCommentBuilderTest {
     }
 
     @Test
-    fun `throws throwable if execution result is null`() {
+    fun `builds empty comment if execution result is null`() {
         val commentBuilder = SimpleClusterCommentBuilder(traceTag, sootToAst)
         val comment = commentBuilder.buildString(sootMethod)
-        val expectedComment = "<pre>\n" +
-                "Test throws Throwable \n" +
-                "</pre>"
-        Assertions.assertEquals(expectedComment, comment)
+        assertEquals(" ", comment)
+    }
+
+    @Test
+    fun `builds empty doc statement if execution result is null`() {
+        val commentBuilder = SimpleClusterCommentBuilder(traceTag, sootToAst)
+        val statements = commentBuilder.buildDocStmts(sootMethod)
+        assertEquals(statements.size, 1)
+        assertEquals(statements[0].toString(), " ")
     }
 
 }

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt
@@ -1,0 +1,57 @@
+package org.utbot.summary.comment
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.utbot.framework.plugin.api.Step
+import org.utbot.framework.plugin.api.UtOverflowFailure
+import org.utbot.summary.ast.JimpleToASTMap
+import org.utbot.summary.tag.StatementTag
+import org.utbot.summary.tag.TraceTag
+import soot.SootMethod
+import soot.jimple.Stmt
+import soot.jimple.internal.JReturnStmt
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SimpleClusterCommentBuilderTest {
+    private lateinit var traceTag: TraceTag
+    private lateinit var jimpleToASTMap: JimpleToASTMap
+    private lateinit var sootToAst: MutableMap<SootMethod, JimpleToASTMap>
+    private lateinit var sootMethod: SootMethod
+    private lateinit var statementTag: StatementTag
+    private lateinit var step: Step
+    private lateinit var statement: Stmt
+
+    @BeforeAll
+    fun setUp() {
+        traceTag = mock(TraceTag::class.java)
+        sootMethod = mock(SootMethod::class.java)
+        jimpleToASTMap = mock(JimpleToASTMap::class.java)
+        statementTag = mock(StatementTag::class.java)
+        step = mock(Step::class.java)
+        statement = mock(JReturnStmt::class.java)
+        sootToAst = mutableMapOf()
+
+        `when`(statementTag.step).thenReturn(step)
+        `when`(step.stmt).thenReturn(statement)
+        `when`(traceTag.path).thenReturn(listOf(step))
+        `when`(traceTag.rootStatementTag).thenReturn(statementTag)
+        `when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
+
+        sootToAst[sootMethod] = jimpleToASTMap
+    }
+
+    @Test
+    fun `throws throwable if execution result is null`() {
+        val commentBuilder = SimpleClusterCommentBuilder(traceTag, sootToAst)
+        val comment = commentBuilder.buildString(sootMethod)
+        val expectedComment = "<pre>\n" +
+                "Test throws Throwable \n" +
+                "</pre>"
+        Assertions.assertEquals(expectedComment, comment)
+    }
+
+}

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
@@ -1,0 +1,57 @@
+package org.utbot.summary.comment
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.utbot.framework.plugin.api.Step
+import org.utbot.framework.plugin.api.UtOverflowFailure
+import org.utbot.summary.ast.JimpleToASTMap
+import org.utbot.summary.tag.StatementTag
+import org.utbot.summary.tag.TraceTag
+import soot.SootMethod
+import soot.jimple.Stmt
+import soot.jimple.internal.JReturnStmt
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SimpleCommentBuilderTest {
+    private lateinit var traceTag: TraceTag
+    private lateinit var jimpleToASTMap: JimpleToASTMap
+    private lateinit var sootToAst: MutableMap<SootMethod, JimpleToASTMap>
+    private lateinit var sootMethod: SootMethod
+    private lateinit var statementTag: StatementTag
+    private lateinit var step: Step
+    private lateinit var statement: Stmt
+
+    @BeforeAll
+    fun setUp() {
+        traceTag = mock(TraceTag::class.java)
+        sootMethod = mock(SootMethod::class.java)
+        jimpleToASTMap = mock(JimpleToASTMap::class.java)
+        statementTag = mock(StatementTag::class.java)
+        step = mock(Step::class.java)
+        statement = mock(JReturnStmt::class.java)
+        sootToAst = mutableMapOf()
+
+        `when`(statementTag.step).thenReturn(step)
+        `when`(step.stmt).thenReturn(statement)
+        `when`(traceTag.path).thenReturn(listOf(step))
+        `when`(traceTag.rootStatementTag).thenReturn(statementTag)
+        `when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
+
+        sootToAst[sootMethod] = jimpleToASTMap
+    }
+
+    @Test
+    fun `throws throwable if execution result is null`() {
+        val commentBuilder = SimpleCommentBuilder(traceTag, sootToAst)
+        val comment = commentBuilder.buildString(sootMethod)
+        val expectedComment = "<pre>\n" +
+                "Test throws Throwable \n" +
+                "</pre>"
+        assertEquals(expectedComment, comment)
+    }
+
+}

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
@@ -64,4 +64,21 @@ class SimpleCommentBuilderTest {
         assertEquals(statements[0].toString(), expectedDocStatement)
     }
 
+    @Test
+    fun `builds inline link for method`() {
+        val commentBuilder = SimpleCommentBuilder(traceTag, sootToAst)
+        val methodReference = commentBuilder.invokeDescription("org.utbot.ClassName", "methodName", listOf())
+        val expectedMethodReference = "{@link org.utbot.ClassName#methodName()}"
+        assertEquals(methodReference, expectedMethodReference)
+    }
+
+    @Test
+    fun `builds inline link for method in nested class`() {
+        val commentBuilder = SimpleCommentBuilder(traceTag, sootToAst)
+        val methodReference =
+            commentBuilder.invokeDescription("org.utbot.ClassName\$NestedClassName", "methodName", listOf())
+        val expectedMethodReference = "{@link org.utbot.ClassName.NestedClassName#methodName()}"
+        assertEquals(methodReference, expectedMethodReference)
+    }
+
 }

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt
@@ -54,4 +54,14 @@ class SimpleCommentBuilderTest {
         assertEquals(expectedComment, comment)
     }
 
+    @Test
+    fun `builds one doc statement`() {
+        val commentBuilder = SimpleCommentBuilder(traceTag, sootToAst)
+        val statements = commentBuilder.buildDocStmts(sootMethod)
+        val expectedDocStatement = "Test \n" +
+                "throws Throwable \n"
+        assertEquals(statements.size, 1)
+        assertEquals(statements[0].toString(), expectedDocStatement)
+    }
+
 }

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
@@ -1,0 +1,38 @@
+package org.utbot.summary.name
+
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.mockito.Mockito
+import org.utbot.framework.plugin.api.UtOverflowFailure
+import org.utbot.summary.ast.JimpleToASTMap
+import org.utbot.summary.tag.TraceTag
+import soot.SootMethod
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SimpleNameBuilderTest {
+    private lateinit var traceTag: TraceTag
+    private lateinit var jimpleToASTMap: JimpleToASTMap
+    private lateinit var sootToAst: MutableMap<SootMethod, JimpleToASTMap>
+    private lateinit var sootMethod: SootMethod
+
+    @BeforeAll
+    fun setUp() {
+        traceTag = Mockito.mock(TraceTag::class.java)
+        Mockito.`when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
+
+        jimpleToASTMap = Mockito.mock(JimpleToASTMap::class.java)
+
+        sootToAst = mutableMapOf()
+        sootMethod = Mockito.mock(SootMethod::class.java)
+        Mockito.`when`(sootMethod.name).thenReturn("methodName")
+        sootToAst[sootMethod] = jimpleToASTMap
+    }
+
+    @Test
+    fun `method name starts with test`() {
+        val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
+        val methodName = simpleNameBuilder.name
+        assert(methodName.startsWith("test"))
+    }
+}

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
@@ -4,7 +4,8 @@ import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
-import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
 import org.utbot.framework.plugin.api.UtOverflowFailure
 import org.utbot.summary.ast.JimpleToASTMap
 import org.utbot.summary.tag.TraceTag
@@ -20,19 +21,19 @@ class SimpleNameBuilderTest {
 
     @BeforeAll
     fun setUp() {
-        traceTag = Mockito.mock(TraceTag::class.java)
-        sootMethod = Mockito.mock(SootMethod::class.java)
-        jimpleToASTMap = Mockito.mock(JimpleToASTMap::class.java)
+        traceTag = mock(TraceTag::class.java)
+        sootMethod = mock(SootMethod::class.java)
+        jimpleToASTMap = mock(JimpleToASTMap::class.java)
         sootToAst = mutableMapOf()
 
-        Mockito.`when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
+        `when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
 
         sootToAst[sootMethod] = jimpleToASTMap
     }
 
     @Test
     fun `method name should start with test`() {
-        Mockito.`when`(sootMethod.name).thenReturn("methodName")
+        `when`(sootMethod.name).thenReturn("methodName")
 
         val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
         val methodName = simpleNameBuilder.name
@@ -41,7 +42,7 @@ class SimpleNameBuilderTest {
 
     @Test
     fun `creates a name pair with a candidate with a bigger index`() {
-        Mockito.`when`(sootMethod.name).thenReturn("")
+        `when`(sootMethod.name).thenReturn("")
 
         val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
         val fromCandidateName = DisplayNameCandidate("fromCandidate", UniquenessTag.Unique, 3)
@@ -57,7 +58,7 @@ class SimpleNameBuilderTest {
 
     @Test
     fun `returns null if candidates are equal`() {
-        Mockito.`when`(sootMethod.name).thenReturn("")
+        `when`(sootMethod.name).thenReturn("")
 
         val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
         val fromCandidateName = DisplayNameCandidate("candidate", UniquenessTag.Unique, 0)

--- a/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
+++ b/utbot-summary/src/test/kotlin/org/utbot/summary/name/SimpleNameBuilderTest.kt
@@ -1,5 +1,6 @@
 package org.utbot.summary.name
 
+import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
@@ -7,6 +8,7 @@ import org.mockito.Mockito
 import org.utbot.framework.plugin.api.UtOverflowFailure
 import org.utbot.summary.ast.JimpleToASTMap
 import org.utbot.summary.tag.TraceTag
+import org.utbot.summary.tag.UniquenessTag
 import soot.SootMethod
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
@@ -19,20 +21,53 @@ class SimpleNameBuilderTest {
     @BeforeAll
     fun setUp() {
         traceTag = Mockito.mock(TraceTag::class.java)
+        sootMethod = Mockito.mock(SootMethod::class.java)
+        jimpleToASTMap = Mockito.mock(JimpleToASTMap::class.java)
+        sootToAst = mutableMapOf()
+
         Mockito.`when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
 
-        jimpleToASTMap = Mockito.mock(JimpleToASTMap::class.java)
-
-        sootToAst = mutableMapOf()
-        sootMethod = Mockito.mock(SootMethod::class.java)
-        Mockito.`when`(sootMethod.name).thenReturn("methodName")
         sootToAst[sootMethod] = jimpleToASTMap
     }
 
     @Test
-    fun `method name starts with test`() {
+    fun `method name should start with test`() {
+        Mockito.`when`(sootMethod.name).thenReturn("methodName")
+
         val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
         val methodName = simpleNameBuilder.name
         assert(methodName.startsWith("test"))
+    }
+
+    @Test
+    fun `creates a name pair with a candidate with a bigger index`() {
+        Mockito.`when`(sootMethod.name).thenReturn("")
+
+        val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
+        val fromCandidateName = DisplayNameCandidate("fromCandidate", UniquenessTag.Unique, 3)
+
+        val toCandidate1 = DisplayNameCandidate("candidate1", UniquenessTag.Common, 2)
+        val toCandidate2 = DisplayNameCandidate("candidate2", UniquenessTag.Common, 4)
+
+        val candidate = simpleNameBuilder.buildCandidate(fromCandidateName, toCandidate1, toCandidate2)
+
+        val resultPair = Pair(fromCandidateName.name, toCandidate2.name)
+        assertEquals(candidate, resultPair)
+    }
+
+    @Test
+    fun `returns null if candidates are equal`() {
+        Mockito.`when`(sootMethod.name).thenReturn("")
+
+        val simpleNameBuilder = SimpleNameBuilder(traceTag, sootToAst, sootMethod)
+        val fromCandidateName = DisplayNameCandidate("candidate", UniquenessTag.Unique, 0)
+
+        // two equal candidates
+        val toCandidate1 = DisplayNameCandidate("candidate", UniquenessTag.Common, 1)
+        val toCandidate2 = DisplayNameCandidate("candidate", UniquenessTag.Common, 1)
+
+        val candidate = simpleNameBuilder.buildCandidate(fromCandidateName, toCandidate1, toCandidate2)
+
+        assertEquals(candidate, null)
     }
 }

--- a/utbot-summary/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
+++ b/utbot-summary/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
@@ -1,0 +1,1 @@
+mock-maker-inline

